From f7c44a4ece6ec90e33bb61080092430d34394569 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Tue, 23 Jun 2015 01:00:35 +0200
Subject: [PATCH 59/70] Add new platformTheme key to themes

Set QT_QPA_PLATFORMTHEME based on what the theme says.
This allows greeter themes to be further customized with special settings.
---
 src/greeter/GreeterApp.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/greeter/GreeterApp.cpp b/src/greeter/GreeterApp.cpp
index cec9d5e..71c7d86 100644
--- a/src/greeter/GreeterApp.cpp
+++ b/src/greeter/GreeterApp.cpp
@@ -107,6 +107,10 @@ namespace SDDM {
         if (m_themeConfig->contains("cursorTheme"))
             qputenv("XCURSOR_THEME", m_themeConfig->value("cursorTheme").toString().toUtf8());
 
+        // set platform theme
+        if (m_themeConfig->contains("platformTheme"))
+            qputenv("QT_QPA_PLATFORMTHEME", m_themeConfig->value("platformTheme").toString().toUtf8());
+
         // create models
 
         m_sessionModel = new SessionModel();
-- 
2.4.3

